### PR TITLE
[Aborea] Create character sheet generation script

### PR DESCRIPTION
**Pull request by Tealk**
# Submission Checklist
## Required

- [x] The pull request title clearly contains the name of the sheet I am editing.
- [x] The pull request title clearly states the type of change I am submitting (New Sheet/New Feature/Bugfix/etc.).
- [x] The pull request makes changes to files in only one sub-folder.
- [x] The pull request does not contain changes to any json files in the translations folder (translation.json is permitted)

# Changes / Description

- Implemented full script to generate the Roll20 character sheet.
- Added functions to generate Spruchlisten HTML, compile Sass to CSS, and combine HTML and JS files into a final output.
- Included logic to update the version in the JSON file if HTML or CSS files have changed.

Since I have seen a Makefile in some sheets, I have now also created a script that builds the character sheet for me, so that errors can be avoided and the simple yaml view makes it easy for inexperienced people to customize the spell list.

### CODE REVIEW

File: Aborea_new/aborea.html [hunk 1]
```diff
@@ -441,1465 +439,1036 @@
...
-          ></button>
+            <input type="checkbox" name="attr_Dauerhafte-Belebung_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:669]: attr_DBelebung_gelernt > attr_Dauerhafte-Belebung_gelernt

File: Aborea_new/aborea.html [hunk 2]
```diff
@@ -441,1465 +439,1036 @@
...
+    </div>
+    <div class="items">
         <div>
-          <button
-            type="roll"
-            name="roll_Erde"
-            value="@{abilityroll} &{template:magie} {{name=Erde}} {{anwendung=-}} {{grundkosten=Grundkosten 8 MP}} {{effekt1=?{Modifikation 1|Elementar mit SG 1 |Elementar mit SG 2 (+ 1 MP) |Elementar mit SG 3 (+ 2 MP) |Elementar mit SG 4 (+ 3 MP) |Elementar mit SG 5 (+ 4 MP) |Elementar mit SG 6 (+ 5 MP) |Elementar mit SG 7 (+ 6 MP) |Elementar mit SG 8 (+ 7 MP) |Elementar mit SG 9 (+ 8 MP) |Elementar mit SG 10 (+ 9 MP) |Elementar mit SG 11 (+ 10 MP) |Elementar mit SG 12 (+ 11 MP) |Elementar mit SG 13 (+ 12 MP) |Elementar mit SG 14 (+ 13 MP) |Elementar mit SG 15 (+ 14 MP) |Elementar mit SG 16 (+ 15 MP) |Elementar mit SG 17 (+ 16 MP) |Elementar mit SG 18 (+ 17 MP) |Elementar mit SG 19 (+ 18 MP) |Elementar mit SG 20 (+ 19 MP) }}} {{effekt2=?{Modifikation 2|1 Stunde |2 Stunden (+1 MP) |3 Stunden (+2 MP) |4 Stunden (+3 MP) |5 Stunden (+1 MP)}}} {{rolling=[[1d10! + @{gezsp_attribut} + @{gezsp} + @{gezsp_bonus} + @{global_modifier} + ?{Modifier?|0}]]}}"
-          ></button>
+            <input type="checkbox" name="attr_Elemente_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:776]: attr_W-Elemente_gelernt > attr_Elemente_gelernt

File: Aborea_new/aborea.html [hunk 3]
```diff
@@ -441,1465 +439,1036 @@
...
-          ></button>
+            <button type="roll" name="roll_Illusion" value="@{abilityroll} &{template:magie} {{name=Illusion}} {{anwendung=Sichtweite}} {{grundkosten=Grundkosten 2 MP}}  {{effekt1=?{Modifikation 1|1 Minute |2 Minuten (+1 MP) |3 Minuten (+2 MP) |4 Minuten (+3 MP) |5 Minuten (+4 MP) |6 Minuten (+5 MP) |7 Minuten (+6 MP) |8 Minuten (+7 MP) |9 Minuten (+8 MP) |10 Minuten (+9 MP) |11 Minuten (+10 MP) |12 Minuten (+11 MP) |13 Minuten (+12 MP) |14 Minuten (+13 MP) |15 Minuten (+14 MP) |16 Minuten (+15 MP) |17 Minuten (+16 MP) |18 Minuten (+17 MP) |19 Minuten (+18 MP) |20 Minuten (+19 MP) |21 Minuten (+20 MP) |22 Minuten (+21 MP) |23 Minuten (+22 MP) |24 Minuten (+23 MP) |25 Minuten (+24 MP) |26 Minuten (+25 MP) |27 Minuten (+26 MP) |28 Minuten (+27 MP) |29 Minuten (+28 MP) |30 Minuten (+29 MP) }}} {{effekt2=?{Modifikation 2||Dauerhaft (aber nicht unzerstörbar) (+10 MP)}}}   {{rolling=[[1d10! + @{gezsp_attribut} + @{gezsp} + @{gezsp_bonus} + @{global_modifier} + ?{Modifier?|0}]]}}"></button>
+        </div>
+    </div>
+    <div class="items">
+        <div>
+            <input type="checkbox" name="attr_Licht-und-Dunkelheit_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:946]: attr_LichtDunkel_gelernt > attr_Licht-und-Dunkelheit_gelernt

File: Aborea_new/aborea.html [hunk 4]
```diff
@@ -441,1465 +439,1036 @@
...
+        </div>
+    </div>
+    <div class="items">
+        <div>
+            <input type="checkbox" name="attr_Grosse-Illusion_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:968]: attr_GIllusion_gelernt > attr_Grosse-Illusion_gelernt

File: Aborea_new/aborea.html [hunk 5]
```diff
@@ -441,1465 +439,1036 @@
...
-          ></button>
+            <input type="checkbox" name="attr_Goettliches-Zeichen_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:1053]: attr_GZeichen_gelernt > attr_Goettliches-Zeichen_gelernt

File: Aborea_new/aborea.html [hunk 6]
```diff
@@ -441,1465 +439,1036 @@
...
+            <button type="roll" name="roll_Weihe" value="@{abilityroll} &{template:magie} {{name=Weihe}} {{anwendung=Berührung}} {{grundkosten=Grundkosten 4 MP}} {{beschraenkungen=ein Tag/Stufe des Formers}} {{effekt1=?{Modifikation 1|3m² |6m² (+1 MP) |9m² (+2 MP) |12m² (+3 MP) |15m² (+4 MP) |18m² (+5 MP) |21m² (+6 MP) |24m² (+7 MP) |27m² (+8 MP) |30m² (+9 MP) }}} {{effekt2=?{Modifikation 2|Feinde des Gottes haben einen Manövermalus -1 |Feinde des Gottes haben einen Manövermalus -2 (+1 MP) |Feinde des Gottes haben einen Manövermalus -3 (+2 MP) |Feinde des Gottes haben einen Manövermalus -4 (+3 MP) |Feinde des Gottes haben einen Manövermalus -5 (+4 MP) |Feinde des Gottes haben einen Manövermalus -6 (+5 MP) |Feinde des Gottes haben einen Manövermalus -7 (+6 MP) |Feinde des Gottes haben einen Manövermalus -8 (+7 MP) }}} {{effekt3=?{Modifikation 3||Der Ort ist als geweihter Ort für alle erkennbar (+3 MP)}}}  {{rolling=[[1d10! + @{gezsp_attribut} + @{gezsp} + @{gezsp_bonus} + @{global_modifier} + ?{Modifier?|0}]]}}"></button>
+        </div>
+    </div>
+    <div class="items">
+        <div>
+            <input type="checkbox" name="attr_Dauerhaftes-zeichen_gelernt" value="1" />
```
Comment (NorWhal) [inline Aborea_new/aborea.html:1086]: attr_DZeichen_gelernt > attr_Dauerhaftes-zeichen_gelernt

**Review by NorWhal — COMMENTED** [review]
It looks like you are changing several attributes that had abbreviated names before. This is going to cause existing sheets to have their data "lost" because the sheet doesn't know that information stored under attributes like `attr_DZeichen_gelernt` don't know that it should now be stored as `attr_Dauerhaftes-zeichen_gelernt`

**Comment by Tealk** [comment]
this is just a nice to have checkbox, the skills of the spells are all still under the skills tab, these are temporary checkboxes that are changed while playing anyway.

**Comment by NorWhal** [comment]
I don't know the game or even the language very well, but it seems like these exist as indicators to mark the spells as "learned?" Players will need to go through and re-mark each spell they knew if this update proceeds as-is.

Maybe "known" spells change from session to session - I'll trust your judgement

**Comment by Tealk** [comment]
the spell list height is listed under 07_fertigkeiten.html, the changes should not cause any problems.

**Pull request merged**